jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+)
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import (
+    PluginTemplateMixin,
+    AddResultsMixin,
+    TableMixin,
+    ViewerSelect,
+    UnitSelectPluginComponent,
+)
+
+__all__ = ["VoPlugin"]
+vo_plugin_label = "Virtual Observatory"
+
+
+@tray_registry("VoPlugin", label=vo_plugin_label)
+class VoPlugin(PluginTemplateMixin, AddResultsMixin, TableMixin):
+    """Plugin to query the Virtual Observatory and load data into Imviz"""
+
+    template_file = __file__, "vo_plugin.vue"
+
+    viewer_items = List([]).tag(sync=True)
+    viewer_selected = Unicode().tag(sync=True)
+
+    wavebands = List().tag(sync=True)
+    waveband_selected = Any().tag(sync=True)  # Any to accept Nonetype
+    resource_filter_coverage = Bool(False).tag(sync=True)
+    resources = List([]).tag(sync=True)
+    resource_selected = Any().tag(sync=True)  # Any to accept Nonetype
+    resources_loading = Bool(False).tag(sync=True)
+
+    source = Unicode("").tag(sync=True)
+    coord_follow_viewer_pan = Bool(False).tag(sync=True)
+    viewer_centered = Bool(False).tag(sync=True)
+    coordframes = List([]).tag(sync=True)
+    coordframe_selected = Unicode("icrs").tag(sync=True)
+    radius_val = Float(1).tag(sync=True)
+    radius_unit_items = List().tag(sync=True)
+    radius_unit_selected = Unicode("deg").tag(sync=True)
+
+    results_loading = Bool(False).tag(sync=True)
+    data_loading = Bool(False).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.viewer = ViewerSelect(
+            self, "viewer_items", "viewer_selected", manual_options=["Manual"]
+        )
+
+        self.radius_unit = UnitSelectPluginComponent(
+            self, items="radius_unit_items", selected="radius_unit_selected"
+        )
+        self.radius_unit.choices = ["deg", "rad", "arcmin", "arcsec"]
+
+        # Waveband properties to filter available registry resources
+        self.wavebands = [
+            w.lower() for w in vocabularies.get_vocabulary("messenger")["terms"]
+        ]
+        self.waveband_selected = None
+
+        self._full_registry_results = None
+        self.resource_selected = None
+
+        self.coordframes = [frame.lower() for frame in all_astropy_frames]
+
+        self.table.headers_avail = ["Title", "Instrument", "DateObs", "URL"]
+        self.table.headers_visible = ["Title", "Instrument", "DateObs"]
+        self._populate_url_only = False
+
+        self.table.show_rowselect = True
+        self.table.item_key = "URL"
+
+        self.hub.subscribe(self, AddDataMessage, handler=self.vue_center_on_data)
+        self.hub.subscribe(self, RemoveDataMessage, handler=self.vue_center_on_data)
+        self.hub.subscribe(self, LinkUpdatedMessage, handler=self.vue_center_on_data)
+
+    @observe("viewer_selected", type="change")
+    def vue_viewer_changed(self, _=None):
+        # Check mixin object initialized
+        if hasattr(self, "viewer"):
```

Comment:
elsewhere we do the following instead which might help readability (whitespace after this will also need updating):

```suggestion
        # Check mixin object initialized
        if not hasattr(self, "viewer"):
            return
```

Reply:
Sure! Done in https://github.com/spacetelescope/jdaviz/pull/2872/commits/3715e4b3db3eeac34ba3674afddadd3544fc4776

Reply:
is it possible to do without passing `event` though?  Can you take out whatever you need from the event and pass as named `kwargs` that the user could pass manually?